### PR TITLE
SPLASH-17/18/19: Redux/Saga/Styled Components Integration

### PR DESCRIPTION
**Pull request by TheLonelyAstronaut**

### CODE REVIEW

File: src/app/main-navigator/home-main-stack.component.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createStackNavigator } from '@react-navigation/stack';
```
Comment (TheLonelyAstronaut) [inline src/app/main-navigator/home-main-stack.component.tsx]: Remove main-navigator folder from commit, we are using routing now

File: src/app/main-navigator/main-stack-param-list.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export type MainStackParamList = {
```
Comment (TheLonelyAstronaut) [inline src/app/main-navigator/main-stack-param-list.tsx]: Same with folder

File: src/app/main-navigator/routes.component.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { NavigationContainer } from '@react-navigation/native';
```
Comment (TheLonelyAstronaut) [inline src/app/main-navigator/routes.component.tsx]: Same

File: src/app/routing/connected-theme-provider.component.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React, { ReactNode } from 'react';
```
Comment (TheLonelyAstronaut) [inline src/app/routing/connected-theme-provider.component.tsx]: Locate this component in the styles folder

File: src/app/main-navigator/tabs-param-list.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export type TabsParamList = {
```
Comment (TheLonelyAstronaut) [inline src/app/main-navigator/tabs-param-list.tsx]: Same

File: src/app/main-navigator/tabs.component.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
```
Comment (TheLonelyAstronaut) [inline src/app/main-navigator/tabs.component.tsx]: Same

File: src/app/routing/root-router.component.tsx [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
 import { AuthStack } from '../authentication/routing';
+import { getIsLoggedIn } from '../authentication/selectors';
 
 import { MainStackComponent } from './main-stack.component';
 
 export const RootRouterComponent: React.FC = () => {
-    const user = 'user';
-    return <NavigationContainer>{user ? <MainStackComponent /> : <AuthStack />}</NavigationContainer>;
+    const isLoggedIn = useSelector(getIsLoggedIn);
+    return <NavigationContainer>{!isLoggedIn ? <MainStackComponent /> : <AuthStack />}</NavigationContainer>;
```
Comment (TheLonelyAstronaut) [inline src/app/routing/root-router.component.tsx]: We need to show MainStack if isLoggedIn === true, so remove ! symbol

File: src/app/styles/actions.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { DefaultTheme } from 'styled-components/native';
+import { createAction } from 'typesafe-redux-helpers';
+
+export interface ThemePayload {
+    theme: DefaultTheme;
+}
+
+export const CHANGE_THEME = createAction('Change Theme', (payload: ThemePayload) => payload);
```
Comment (TheLonelyAstronaut) [inline src/app/styles/actions.ts]: Rewrite string in createAction to '[Change Theme]' - naming convention

File: src/app/styles/login.ts [hunk 9]
```diff
@@ -0,0 +1,97 @@
+import styled from 'styled-components/native';
+
+export const Container = styled.View`
+    background-color: ${(props) => props.theme.colors.main};
+    flex: 1;
+`;
+export const Title = styled.Text`
+    color: ${(props) => props.theme.colors.secondary};
+    font-size: 48px;
```
Comment (TheLonelyAstronaut) [inline src/app/styles/login.ts]: All numbers should be in the Default Theme interface and then we can use it like here:

font-size: ${(props) => props.theme.text.header}px;

File: src/app/routing/root-router.component.tsx (hunk 7 above)
Comment (Pepegaga) [reply src/app/routing/root-router.component.tsx]: That was just to test login screen

**Review by TheLonelyAstronaut — COMMENTED** [review]
Great work

**Pull request merged**